packages/wallets/src/evm/connectors/smart-wallet/lib/erc4337-signer.ts

```diff
@@ -139,17 +139,18 @@ Code: ${errorCode}`;
   }
 
   async signMessage(message: Bytes | string): Promise<string> {
-    const isNotDeployed = await this.smartAccountAPI.checkAccountPhantom();
-    if (isNotDeployed) {
-      console.log(
-        "Account contract not deployed yet. Deploying account before signing message",
-      );
-      const tx = await this.sendTransaction({
-        to: await this.getAddress(),
-        data: "0x",
-      });
-      await tx.wait();
-    }
+      const isNotDeployed = await this.smartAccountAPI.checkAccountPhantom();
+      if (isNotDeployed && (this.config.doNotDeployOnSignMessage === undefined || this.config.doNotDeployOnSignMessage === false)) {
```

Comment:
can just do !varName

Reply:
kk

Reply:
fixed in https://github.com/thirdweb-dev/js/pull/2222/commits/6226a1973a492ee2a2369333f8c8d13bf8414812